tutordiscovery/plugin.py

```diff
@@ -29,18 +31,86 @@
     },
 }
 
-hooks = {
-    "build-image": {"discovery": "{{ DISCOVERY_DOCKER_IMAGE }}"},
-    "remote-image": {"discovery": "{{ DISCOVERY_DOCKER_IMAGE }}"},
-    "init": ["mysql", "lms", "discovery"],
-}
+# Initialization tasks
+tutor_hooks.Filters.COMMANDS_INIT.add_item(
+    (
+        "mysql",
+        ("discovery", "tasks", "mysql", "init"),
+    )
+)
+tutor_hooks.Filters.COMMANDS_INIT.add_item(
+    (
+        "lms",
+        ("discovery", "tasks", "lms", "init"),
+    )
+)
+tutor_hooks.Filters.COMMANDS_INIT.add_item(
+    (
+        "discovery",
+        ("discovery", "tasks", "discovery", "init"),
+    )
+)
+
+# Image management
+tutor_hooks.Filters.IMAGES_BUILD.add_item(
+    (
+        "discovery",
+        ("plugins", "discovery", "build", "discovery"),
+        "{{ DISCOVERY_DOCKER_IMAGE }}",
+        (),
+    )
+)
+tutor_hooks.Filters.IMAGES_PULL.add_item(
+    (
+        "discovery",
+        "{{ DISCOVERY_DOCKER_IMAGE }}",
+    )
+)
+tutor_hooks.Filters.IMAGES_PUSH.add_item(
+    (
+        "discovery",
+        "{{ DISCOVERY_DOCKER_IMAGE }}",
+    )
+)
+
+# Automount /openedx/discovery folder from the container
+@tutor_hooks.Filters.COMPOSE_MOUNTS.add()
+def _mount_course_discovery(mounts, name):
+    if name == "course-discovery":
+        mounts.append(("discovery", "/openedx/discovery"))
+    return mounts
 
 
-def patches():
-    all_patches = {}
-    for path in glob(os.path.join(HERE, "patches", "*")):
-        with open(path) as patch_file:
-            name = os.path.basename(path)
-            content = patch_file.read()
-            all_patches[name] = content
-    return all_patches
+####### Boilerplate code
+# Add the "templates" folder as a template root
+tutor_hooks.Filters.ENV_TEMPLATE_ROOTS.add_item(
+    pkg_resources.resource_filename("tutordiscovery", "templates")
+)
+# Render the "build" and "apps" folders
+tutor_hooks.Filters.ENV_TEMPLATE_TARGETS.add_items(
+    [
+        ("discovery/build", "plugins"),
+        ("discovery/apps", "plugins"),
+    ],
+)
+# Load patches from files
+for path in glob(
+    os.path.join(
+        pkg_resources.resource_filename("tutordiscovery", "patches"),
+        "*",
+    )
+):
+    with open(path, encoding="utf-8") as patch_file:
+        tutor_hooks.Filters.ENV_PATCHES.add_item(
+            (os.path.basename(path), patch_file.read())
+        )
+# Add configuration entries
+tutor_hooks.Filters.CONFIG_DEFAULTS.add_items(
+    [(f"DISCOVERY_{key}", value) for key, value in config.get("defaults", {}).items()]
+)
+tutor_hooks.Filters.CONFIG_UNIQUE.add_items(
+    [(f"DISCOVERY_{key}", value) for key, value in config.get("unique", {}).items()]
+)
+tutor_hooks.Filters.CONFIG_OVERRIDES.add_items(
+    list(config.get("overrides", {}).items())
+)
```

Comment:
Is this necessary since the `config` dict above doesn't have `"overrides"`?

Reply:
No, but it might be in the future. It's just a standard boilerplate.